### DISCUSSION TITLE
Private GitHub NPM packages with new Host Rule feature

### DISCUSSION THREAD

**Discussion by sebastianbuechler** [renovatebot/renovate#32079] [Request Help]
### How are you running Renovate?

A Mend.io-hosted app

### Please tell us more about your question or problem

I wanted to setup the access to private NPM packages hosted on GitHub via the new feature of specifying the host rules in the developer UI of mend.io:
![image](https://github.com/user-attachments/assets/cd113d73-3049-4990-baa2-d343408f650f)
I can see in the debug messages that this rule gets picked up:
`Adding token authentication for https://npm.pkg.github.com/ (hostType=npm) to hostRules`
But when it should install my packages it fails because it seems to fetch it through https://registry.npmjs.org and not the specified https://npm.pkg.github.com/:
```
DEBUG: GET https://registry.npmjs.org/@<myOrg>%2F>my_package> = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=404 retryCount=0, duration=2592)
DEBUG: Failed to look up npm package @<myOrg>/<my_package>
```

Shouldn't there be an option to specify the registry I want to use for package of my organization(like before it was done with npmrc)? 
I'm using bun as package manager.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by RahulGautamSingh** [comment]
Change the resgistryUrl for npm datasource (cause bun uses this) to the one you want using package rules.

The lockfile updates will still use the npm registry though because we do not yet have npmrc support for bun [^1]

[^1]: https://github.com/renovatebot/renovate/issues/30926
